### ISSUE TITLE
New cask: GitHub fork (not canonical repository)

### ISSUE THREAD

**Issue by d029940** [Homebrew/homebrew-cask#142510]
### Verification

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [X] I have retried my command with `--force`.
- [X] I ran `brew update-reset && brew update` and retried my command.
- [X] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [X] I made doubly sure this is not a [checksum does not match](https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match) error.

### Description of issue

I'm a newbie and I'm about to create a cask for an app called QMapShack.
There has been a cask for this app [Delete qmapshack.rb](https://github.com/Homebrew/homebrew-cask/pull/69282) (#69282), but it is abandoned. So, I will take over.

I followed the instructions on [Getting Set Up To Contribute](https://github.com/d029940/homebrew-cask/blob/master/CONTRIBUTING.md#getting-set-up-to-contribute) where the 4. step is `git checkout -b qmapshack`.

I followed the instructions in [Adding Software To Homebrew](https://docs.brew.sh/Adding-Software-to-Homebrew).

1. I created `qmapshack.rb`
2. I ran `brew install qmapshack`
3. I ran `brew uninstall qmapshack`
4. I ran `brew audit --new-cask qmapshack` and got the error message `GitHub fork (not canonical repository)`. I have no idea as to why.
5. I ran `brew style --fix qmapshack` without any errors.

The app itself is located at [QMapShack for the Mac](https://github.com/d029940/qmapshack/releases/tag/QMapShack-Mac).

Because or the error in `brew audit --new-cask qmapshack`, I didn't submit the cask.

So, what is wrong with "GitHub fork (not canonical repository)". Should I submit it anyway?

### Command that failed

brew audit --new-cask qmapshack

### Output of command with `--verbose --debug`

```shell
==> Downloading https://github.com/d029940/qmapshack/releases/download/QMapShack-Mac/QMapShack_1.16.1-2-arm-brew.zip
Already downloaded: /Users/manfred/Library/Caches/Homebrew/downloads/4706a8704656072445f196d19239abf1cebfbede1ef475f1dc368dc24831547c--QMapShack_1.16.1-2-arm-brew.zip
Warning: No checksum defined for cask 'qmapshack', skipping verification.
==> Downloading https://github.com/d029940/qmapshack/releases/download/QMapShack-Mac/QMapShack_1.16.1-2-arm-brew.zip
Already downloaded: /Users/manfred/Library/Caches/Homebrew/downloads/4706a8704656072445f196d19239abf1cebfbede1ef475f1dc368dc24831547c--QMapShack_1.16.1-2-arm-brew.zip
Warning: No checksum defined for cask 'qmapshack', skipping verification.
audit for qmapshack: failed
 - GitHub fork (not canonical repository)
Error: 1 problem in 1 cask detected
```

### Output of `brew doctor` and `brew config`

```shell
brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have uncommitted modifications to Homebrew/homebrew-cask.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /opt/homebrew/Library/Taps/homebrew/homebrew-cask && git stash -u && git clean -d -f

Uncommitted files:
  ?? Casks/qmapshack.rb

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/cask) checkout master

brew config:
HOMEBREW_VERSION: 4.0.5-92-gfdd7b04
ORIGIN: https://github.com/Homebrew/brew
HEAD: fdd7b04ccf14041e9ccd794daa966ee4db5ef2eb
Last commit: 2 hours ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 41dfd4d18b29b4ee9921aec7daaacbc55723c170
Core tap last commit: 2 hours ago
Core tap branch: master
Core tap JSON: 07 Mar 16:56 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.SKXTZ2WkHZ/org.xquartz:0
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.39.2 => /opt/homebrew/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew tap`

```shell
homebrew/cask
homebrew/core
sidneys/homebrew
```
[qmapshack.rb.zip](https://github.com/Homebrew/homebrew-cask/files/10912494/qmapshack.rb.zip)

**Comment by SMillerDev** [comment]
> So, what is wrong with "GitHub fork (not canonical repository)". Should I submit it anyway?

Forks are expected to submit their changes upstream so we should not distribute from forks. If you have no intention of doing that, it would be better to unfork the repo to indicate that to people

**Comment by d029940** [comment]
Ok, this is a restriction, where I probably will not create a brew cask.

The situation: The Upstream of QMapShack mainly focuses on Windows and Linux. So, my fork of QMapShack does the build for MacOS and stores the executable in my release folder. All the changes of the build scripts go to upstream, but not the executables.
I am not sure what uncorking really does, but I would expect, that I do not get any change notification in my fork if there are changes in upstream. If so, that is a no go. I only build for MacOS.

To sum up:
- Unforking will cut the connection of my fork to upstream. Right?
- Building MacOS bundles with my fork will not go to upstream (only source changes)

Conclusion: No brew package. I will offer the MacOS bundle in my fork as a release.

If my understanding is correct, we can close this issue.

**Comment by SMillerDev** [comment]
> Building MacOS bundles with my fork will not go to upstream (only source changes)

Yeah, since there doesn't seem to be any official relation between the upstream project and your fork I don't think it would be accepted here.

You could host it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Comment by d029940** [comment]
Thx. I will have a look at "personal tap"

**Comment by cmeid** [comment]
Did you have a chance to look at that @d029940? I would love to use your formula!

**Comment by d029940** [comment]
I stopped working on a brew package.

There are 2 possibilities:

1. Create a brew tap OR
2. Create a separate GitHub project unrelated to the original QMapShack from kiozen (like so [https://github.com/d029940/qmapshack-brew/blob/main/qmapshack.rb](https://github.com/d029940/qmapshack-brew/blob/main/qmapshack.rb)

If anyone likes to take over the creation of a brew package, I will appreciate it very much. Currently, my time is very constrained.

Also, the work branch of [https://github.com/d029940/qmapshack](https://github.com/d029940/qmapshack) has the latest build scripts. I am using these build scripts on my local machine.

If there are any questions, don't hesitate.
